feat(express): Introduce Clerk package for Express (#2918)
clerk/javascript: .changeset/bright-gorillas-crash.md

---
'@clerk/express': patch
---

> Introduce Express specific Clerk SDK `@clerk/express`. The SDK exposes the following API:
> 
> - clerkClient: default @clerk/backend client initialized from env variables and used to make BAPI requests
> - clerkMiddleware: centralized middleware that authenticates all requests without blocking them (also triggers handshake mechanism)
> - getAuth: utility to retrieve the auth state from the request (requires clerkMiddleware to executed)
> - protect: middleware that returns http 401 response when request is signed-out
> 
> Also all the top level exports from `@clerk/backend` are re-exported from `@clerk/express` top level.

Reviewer: ```suggestion
Introduce [Express](https://expressjs.com/) specific Clerk SDK called `@clerk/express`. The SDK exposes the following APIs:

- `clerkClient`: Default [`@clerk/backend`](https://clerk.com/docs/references/backend/overview) client initialized from environment variables and used to make backend API requests
- `clerkMiddleware`: Centralized middleware that authenticates all requests without blocking them (also triggers handshake mechanism)
- `getAuth`: Utility to retrieve the auth state from a request (requires `clerkMiddleware` to be executed)
- `protect`: Middleware that returns HTTP 401 response when request is signed-out

Also all the top level exports from `@clerk/backend` are re-exported from `@clerk/express`.
```
